nucliadb_node/src/grpc/grpc_writer.rs

```diff
@@ -224,20 +240,24 @@ impl NodeWriter for NodeWriterGRPCDriver {
         let shard_id = resource.shard_id.clone();
         let shard = self.obtain_shard(&shard_id).await?;
         let info = info_span!(parent: &span, "remove resource");
-        let task = || {
+        let merge_task = create_merge_task(shard.clone());
+        let write_task = || {
             run_with_telemetry(info, move || {
                 shard
                     .remove_resource(&resource)
                     .and_then(|()| shard.get_opstatus())
             })
         };
-        let status = tokio::task::spawn_blocking(task).await.map_err(|error| {
-            tonic::Status::internal(format!("Blocking task panicked: {error:?}"))
-        })?;
+        let status = tokio::task::spawn_blocking(write_task)
+            .await
+            .map_err(|error| {
+                tonic::Status::internal(format!("Blocking task panicked: {error:?}"))
+            })?;
         match status {
             Ok(mut status) => {
                 status.status = 0;
                 status.detail = "Success!".to_string();
+                tokio::task::spawn_blocking(merge_task);
```

Comment:
I am confused, are you returning two statuses here?

Reply:
no! one status and I spawn a task that checks if I can merge